packages/next/src/build/noop-react-dom-server-legacy.ts

```diff
@@ -0,0 +1,11 @@
+export function renderToString() {
+  throw new Error(
+    `renderToString() is not supported when importing 'react-dom/server.edge' or 'react-dom/server.browser', use 'react-dom/server' instead.`
```

Comment:
This message is also confusing b/c it seems like this is something React doesn't support when it's just a restriction Next is putting in place.

To do this right we're going to need to do something more dramatic I think. One idea is to rewrite imports of `react-dom/server` so that the parts using modern imports point to the runtime and the parts using legacy methods bundle the legacy builds (which also point to the runtime for it's dependence on `react` and `react-dom`. This would potentially require splitting a single import into multiple and I'm not really sure how to do that but my guess is it's possible with our swc loader.

Seems like a heavy lift though.